src/main/java/org/sagebionetworks/web/client/widget/discussion/modal/NewDiscussionThreadModal.java

```diff
@@ -71,25 +70,37 @@ public void onSave() {
 		discussionForumClient.createThread(toCreate, new AsyncCallback<DiscussionThreadBundle>(){
 			@Override
 			public void onFailure(Throwable caught) {
-				// TODO Auto-generated method stub
-				
+				synAlert.handleException(caught);
 			}
 
 			@Override
 			public void onSuccess(DiscussionThreadBundle result) {
+				view.hideDialog();
+				view.showSuccess();
 				if (newThreadCallback != null) {
-					newThreadCallback.invoke(null);
+					newThreadCallback.invoke();
 				}
 			}
 		});
 	}
 
-	private boolean isValidMessage(String messageMarkdown) {
-		return (messageMarkdown != null && !messageMarkdown.equals(""));
-	}
-
-	private boolean isValidTitle(String threadTitle) {
-		return (threadTitle != null && !threadTitle.equals(""));
+	public static ValidationResult validate(String threadTitle, String messageMarkdown) {
+		String errorMessages = "";
+		ValidationResult result = new ValidationResult();
+		result.setValidity(true);
+		if (threadTitle == null || threadTitle.equals("")) {
```

Comment:
That sounds great! I will do that. In what package should the ValidationResult class be?